src/compiler/crystal/interpreter/compiler.cr

```diff
@@ -1871,7 +1871,7 @@ class Crystal::Repl::Compiler < Crystal::Visitor
                    begin
                      create_compiled_def(node, target_def)
                    rescue ex : Crystal::TypeException
-                     node.raise ex, inner: ex
+                     node.raise ex.message, inner: ex
```

Comment:
I think manually creating the type exceptions via `.for_node` changed something that caused these calls to be taken into consideration and error due to the first arg expecting a string not the exception instance.